src/JsonSchema/Constraints/Factory.php

```diff
@@ -82,6 +87,23 @@ public function __construct(
     }
 
     /**
+     * Get package files
+     *
+     * @param string File path, relative to the root of the package
+     */
+    public function getPackageFile($path)
```

Comment:
Can you clarify what you think it's duplicating? I don't see anything in`SchemaStorage` that would allow me to achieve the same result - as far as I can see, the only similarity is that they both cache stuff.